pkg/controller/seed-controller-manager/etcdbackup/etcd_backup_controller.go

```diff
@@ -0,0 +1,983 @@
+	mgr manager.Manager,
+	log *zap.SugaredLogger,
+	numWorkers int,
+	workerName string,
+	storeContainer *corev1.Container,
+	deleteContainer *corev1.Container,
+	cleanupContainer *corev1.Container,
+	backupContainerImage string,
+) error {
+	log = log.Named(ControllerName)
+	client := mgr.GetClient()
+
+	if backupContainerImage == "" {
+		backupContainerImage = DefaultBackupContainerImage
+	}
+
+	reconciler := &Reconciler{
+		log:                  log,
+		Client:               client,
+		workerName:           workerName,
+		storeContainer:       storeContainer,
+		deleteContainer:      deleteContainer,
+		cleanupContainer:     cleanupContainer,
+		backupContainerImage: backupContainerImage,
+		recorder:             mgr.GetEventRecorderFor(ControllerName),
+		clock:                &clock.RealClock{},
+		randStringGenerator: func() string {
+			return rand.String(10)
+		},
+	}
+
+	ctrlOptions := controller.Options{
+		Reconciler:              reconciler,
+		MaxConcurrentReconciles: numWorkers,
+	}
+	c, err := controller.New(ControllerName, mgr, ctrlOptions)
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(&source.Kind{Type: &kubermaticv1.EtcdBackupConfig{}}, &handler.EnqueueRequestForObject{})
+}
+
+// Reconcile handle etcd backups reconciliation.
+func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	log := r.log.With("request", request)
+	log.Debug("Processing")
+
+	backupConfig := &kubermaticv1.EtcdBackupConfig{}
+	if err := r.Get(ctx, request.NamespacedName, backupConfig); err != nil {
+		if kerrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	cluster := &kubermaticv1.Cluster{}
+	if err := r.Get(ctx, types.NamespacedName{Name: backupConfig.Spec.Cluster.Name}, cluster); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	log = r.log.With("cluster", cluster.Name, "backupConfig", backupConfig.Name)
+
+	// Add a wrapping here so we can emit an event on error
+	result, err := kubermaticv1helper.ClusterReconcileWrapper(
+		ctx,
+		r.Client,
+		r.workerName,
+		cluster,
+		kubermaticv1.ClusterConditionBackupControllerReconcilingSuccess,
+		func() (*reconcile.Result, error) {
+			return r.reconcile(ctx, log, backupConfig, cluster)
+		},
+	)
+	if err != nil {
+		log.Errorw("Reconciling failed", zap.Error(err))
+		r.recorder.Event(backupConfig, corev1.EventTypeWarning, "ReconcilingError", err.Error())
+		r.recorder.Eventf(cluster, corev1.EventTypeWarning, "ReconcilingError",
+			"failed to reconcile etcd backup config %q: %v", backupConfig.Name, err)
+	}
+	if result == nil {
+		result = &reconcile.Result{}
+	}
+	return *result, err
+}
+
+func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, backupConfig *kubermaticv1.EtcdBackupConfig, cluster *kubermaticv1.Cluster) (*reconcile.Result, error) {
+	if err := r.ensureEtcdClientSecret(ctx, cluster); err != nil {
```

Comment:
Why do we need this ?

Reply:
I copied this from the old backup controller. At least one controller needs to do it I figure.